### ISSUE TITLE
support HTTP timeouts (or other request options)

### ISSUE THREAD

**Issue by jbcpollak** [strongloop/loopback-connector-remote#48]
I would like to be able to control the request timeout from the client side, but it appears `loopback-connector-remote` does not expose this functionality - It looks like `strong-remoting` doesn't allow the user to add fields to the `request` options object, that might be by design, but I think it would be useful to expose `timeout` and possibly a few other fields.

The lack of a timeout is causing me problems right now - the remote call doesn't timeout in a reasonable time, and my unit test actually times out before the call does, so I couldn't even see the root cause of the error, just that the test died. Now that I've figured that out, I'd like to set my timeout to be ~5 seconds (unit test timeout is 15 seconds), so that in the future this would be catchable.

**Comment by richardpringle** [comment]
@jbcpollak, would you want this feature to be `connector-remote` specific? Or do you think it should be a part of strong-remoting? I think the change would have to be made in strong-remoting, might be better to open up an issue in there or we can leave this issue open and cross-link the two.

Open a new issue in [strong-remoting](https://github.com/strongloop/strong-remoting/issues) and mention me in the description please.

**Comment by jbcpollak** [comment]
As an end user I'm interested in the feature being exposed here, but I do agree it should probably be implemented in strong-remoting. Honestly as a user of Loopback, I don't really get what the utility of strong-remoting is outside of Loopback.

**Comment by jbcpollak** [comment]
I still think this is important and shouldn't be closed.

**Comment by richardpringle** [comment]
@kjdelisle, calm that stalebot down! haha I'm with @jbcpollak on this one, I think this feature shoud be in here.

**Comment by bajtos** [comment]
I agree we should support timeouts, I think we should support [Circuit Breaker pattern](https://martinfowler.com/bliki/CircuitBreaker.html) too.

Unfortunately, we don't have bandwidth to implement this feature ourselves. Would anybody watching this issue like to contribute this feature?

Having open too many issues that will never be implemented/fixed makes it difficult for us to stay on top of our ever-growing backlog. The stalebot is automatically closing issues with no activity, to keep the size of our backlog more reasonable.

**Comment by regevbr** [comment]
I have many issues due to this missing feature - a service starts a long request to a remote service and mid way the remote service gets shutdown. Currently the request hangs forever causing leaks and undesired behavior of the application - espceially when the operation is an async non user facing operation where there are no timeouts set up regardless of the remote operation
